### DISCUSSION TITLE
Delete user based on column value

### DISCUSSION THREAD

**Discussion by scottklein7** [supabase/supabase#21648] [Questions] [answered]
Hi all!

I have a table that stores user data like when the subscription was created and if the date the user cancelled the subscription (if they did). 
This is the problem I am trying to solve: 
Say a user buys a subscription on march 2 and has 30 day access, within the subscription created at column it then stores that date as March 2 2024. On March 15 they decide to cancel their subscription but will still have access until april 2nd, upon cancelling a column called canceled_at is updated to March 15. After april 2nd I need to deltete the user from that table and the auth table. Therfore, what would be the best approach? A cron job? Severless function? What would be the best approach.

Thanks a lot!

**Comment by GaryAustin1** [comment]
Cron seems ideal.

**Reply by scottklein7** [reply]
Would a daily cron be best?

**Reply by GaryAustin1** [reply, marked as answer]
I would run a cron task with the lowest resolution acceptable to you and balancing how many deletes might need to occur.  Daily or hourly seems sufficient for such a non critical point in time.

**Reply by scottklein7** [reply]
Awesome, thanks very much!

**Comment by scottklein7** [comment]
Hey all, would this be a good approach? Is there no way to batch delete users in supabase? This would be a daily cron job running on vercel. 

```
import { Database } from '@/types_db';
import { createClient } from '@supabase/supabase-js';
import { NextResponse } from 'next/server';

export const dynamic = 'force-dynamic';

export async function GET() {
  try {
    const supabase = createClient<Database>(
      process.env.NEXT_PUBLIC_SUPABASE_URL || '',
      process.env.SUPABASE_SERVICE_ROLE_KEY || ''
    );

    const thirtyDaysAgo = new Date();
    thirtyDaysAgo.setDate(thirtyDaysAgo.getDate() - 30);
    thirtyDaysAgo.setHours(0, 0, 0, 0); // Normalize time part if needed

    // Adjust the query to fetch only users who meet the specific criteria
    const { data, error } = await supabase
      .from('kajabi')
      .select('*')
      .gt('sub_created_at', thirtyDaysAgo.toISOString()) // Use `gt` for greater than
      .neq('canceled_at', null);

    if (error) throw error;

    if (data) {
      for (const user of data) {
        if (user.id) {
          const { error: deleteUserError } =
            await supabase.auth.admin.deleteUser(user.id);
          if (deleteUserError) throw deleteUserError;
        }
      }
    }

    // If the process is successful
    return NextResponse.json({ status: 200, message: 'Success' });
  } catch (error) {
    // Handle any errors that occurred during the process
    return NextResponse.json({
      status: 400,
      message: `An error occurred: ${JSON.stringify(error)}`
    });
  }
}
```
